File: ios/runanywhere-sdks-latest/examples/flutter/RunAnywhereAI/lib/core/design_system/app_spacing.dart
/// App Spacing (mirroring iOS AppSpacing.swift)
class AppSpacing {
  // MARK: - Padding values
  static const double xxSmall = 2;
  static const double xSmall = 4;
  static const double small = 6;
  static const double smallMedium = 8;
  static const double medium = 10;
  static const double mediumLarge = 12;
  static const double regular = 14;
  static const double large = 16;
  static const double xLarge = 20;
  static const double xxLarge = 30;
  static const double xxxLarge = 40;

  // MARK: - Specific padding values
  static const double padding4 = 4;
  static const double padding6 = 6;
  static const double padding8 = 8;
  static const double padding9 = 9;
  static const double padding10 = 10;
  static const double padding12 = 12;
  static const double padding14 = 14;
  static const double padding15 = 15;
  static const double padding16 = 16;
  static const double padding20 = 20;
  static const double padding24 = 24;
  static const double padding30 = 30;
  static const double padding32 = 32;
  static const double padding40 = 40;
  static const double padding60 = 60;
  static const double padding100 = 100;

  // MARK: - Icon sizes
  static const double iconSmall = 8;
  static const double iconRegular = 18;
  static const double iconMedium = 28;
  static const double iconLarge = 48;
  static const double iconXLarge = 60;
  static const double iconXXLarge = 72;
  static const double iconHuge = 80;

  // MARK: - Button sizes
  static const double buttonHeightSmall = 28;
  static const double buttonHeightRegular = 44;
  static const double buttonHeightLarge = 72;

  // MARK: - Corner radius
  static const double cornerRadiusSmall = 4;
  static const double cornerRadiusMedium = 6;
  static const double cornerRadiusRegular = 8;
  static const double cornerRadiusLarge = 10;
  static const double cornerRadiusXLarge = 12;
  static const double cornerRadiusXXLarge = 14;
  static const double cornerRadiusCard = 16;
  static const double cornerRadiusBubble = 18;
  static const double cornerRadiusModal = 20;

  // MARK: - Frame sizes
  static const double minFrameHeight = 150;
  static const double maxFrameHeight = 150;

  // MARK: - Stroke widths
  static const double strokeThin = 0.5;
  static const double strokeRegular = 1.0;
  static const double strokeMedium = 2.0;

  // MARK: - Shadow radius
  static const double shadowSmall = 2;
  static const double shadowMedium = 3;
  static const double shadowLarge = 4;
  static const double shadowXLarge = 10;
}

/// Layout Constants (mirroring iOS AppLayout)
class AppLayout {
  // MARK: - macOS specific (for desktop Flutter)
  static const double macOSMinWidth = 400;
  static const double macOSIdealWidth = 600;
  static const double macOSMaxWidth = 900;
  static const double macOSMinHeight = 300;
  static const double macOSIdealHeight = 500;
  static const double macOSMaxHeight = 800;

  // MARK: - Content width limits
  static const double maxContentWidth = 800;
  static const double maxContentWidthLarge = 1000;
  static const double maxContentWidthXLarge = 1200;

  // MARK: - Sheet sizes
  static const double sheetMinWidth = 500;
  static const double sheetIdealWidth = 600;
  static const double sheetMaxWidth = 700;
  static const double sheetMinHeight = 400;
  static const double sheetIdealHeight = 500;
  static const double sheetMaxHeight = 600;

  // MARK: - Animation durations
  static const Duration animationFast = Duration(milliseconds: 250);
  static const Duration animationRegular = Duration(milliseconds: 300);
  static const Duration animationSlow = Duration(milliseconds: 500);
  static const Duration animationVerySlow = Duration(milliseconds: 600);
  static const Duration animationLoop = Duration(seconds: 1);
  static const Duration animationLoopSlow = Duration(seconds: 2);
}

File: ios/runanywhere-sdks-latest/examples/ios/RunAnywhereAI/RunAnywhereAI/Core/DesignSystem/AppSpacing.swift
//
//  AppSpacing.swift
//  RunAnywhereAI
//
//  Centralized spacing and sizing from existing usage in the app
//

import SwiftUI

// MARK: - App Spacing (gathered from existing usage)
struct AppSpacing {
    // MARK: - Padding values (from existing usage)
    static let xxSmall: CGFloat = 2
    static let xSmall: CGFloat = 4
    static let small: CGFloat = 6
    static let smallMedium: CGFloat = 8
    static let medium: CGFloat = 10
    static let mediumLarge: CGFloat = 12
    static let regular: CGFloat = 14
    static let large: CGFloat = 16
    static let xLarge: CGFloat = 20
    static let xxLarge: CGFloat = 30
    static let xxxLarge: CGFloat = 40

    // Specific padding values used
    static let padding4: CGFloat = 4
    static let padding6: CGFloat = 6
    static let padding8: CGFloat = 8
    static let padding9: CGFloat = 9
    static let padding10: CGFloat = 10
    static let padding12: CGFloat = 12
    static let padding14: CGFloat = 14
    static let padding15: CGFloat = 15
    static let padding16: CGFloat = 16
    static let padding20: CGFloat = 20
    static let padding30: CGFloat = 30
    static let padding40: CGFloat = 40
    static let padding60: CGFloat = 60
    static let padding100: CGFloat = 100

    // MARK: - Component sizes (from existing usage)

    // Icon sizes
    static let iconSmall: CGFloat = 8
    static let iconRegular: CGFloat = 18
    static let iconMedium: CGFloat = 28
    static let iconLarge: CGFloat = 48
    static let iconXLarge: CGFloat = 60
    static let iconXXLarge: CGFloat = 72
    static let iconHuge: CGFloat = 80

    // Button sizes
    static let buttonHeightSmall: CGFloat = 28
    static let buttonHeightRegular: CGFloat = 44
    static let buttonHeightLarge: CGFloat = 72

    // Corner radius (from existing usage)
    static let cornerRadiusSmall: CGFloat = 4
    static let cornerRadiusMedium: CGFloat = 6
    static let cornerRadiusRegular: CGFloat = 8
    static let cornerRadiusLarge: CGFloat = 10
    static let cornerRadiusXLarge: CGFloat = 12
    static let cornerRadiusXXLarge: CGFloat = 14
    static let cornerRadiusCard: CGFloat = 16
    static let cornerRadiusBubble: CGFloat = 18
    static let cornerRadiusModal: CGFloat = 20

    // Frame sizes (from existing usage)
    static let minFrameHeight: CGFloat = 150
    static let maxFrameHeight: CGFloat = 150

    // Stroke widths
    static let strokeThin: CGFloat = 0.5
    static let strokeRegular: CGFloat = 1.0
    static let strokeMedium: CGFloat = 2.0

    // Shadow radius
    static let shadowSmall: CGFloat = 2
    static let shadowMedium: CGFloat = 3
    static let shadowLarge: CGFloat = 4
    static let shadowXLarge: CGFloat = 10
}

// MARK: - Layout Constants (from existing usage)
struct AppLayout {
    // macOS specific
    static let macOSMinWidth: CGFloat = 400
    static let macOSIdealWidth: CGFloat = 600
    static let macOSMaxWidth: CGFloat = 900
    static let macOSMinHeight: CGFloat = 300
    static let macOSIdealHeight: CGFloat = 500
    static let macOSMaxHeight: CGFloat = 800

    // Content width limits
    static let maxContentWidth: CGFloat = 800
    static let maxContentWidthLarge: CGFloat = 1000
    static let maxContentWidthXLarge: CGFloat = 1200

    // Sheet sizes
    static let sheetMinWidth: CGFloat = 500
    static let sheetIdealWidth: CGFloat = 600
    static let sheetMaxWidth: CGFloat = 700
    static let sheetMinHeight: CGFloat = 400
    static let sheetIdealHeight: CGFloat = 500
    static let sheetMaxHeight: CGFloat = 600

    // Animation durations
    static let animationFast: Double = 0.25
    static let animationRegular: Double = 0.3
    static let animationSlow: Double = 0.5
    static let animationVerySlow: Double = 0.6
    static let animationLoop: Double = 1.0
    static let animationLoopSlow: Double = 2.0
}
